src/Summoner/Text.hs

```diff
@@ -14,8 +14,8 @@ packageToModule = T.concat . map headToUpper . T.splitOn "-"
 
 -- | Converts every element of list into 'Text' and then joins every element
 -- into single 'Text' like 'T.intercalate'.
-intercalateMap :: Text -> (a -> Text) -> [a] -> Text
-intercalateMap between showT = T.intercalate between . map showT
+intercalateMap :: (Functor t, Foldable t) => Text -> (a -> Text) -> t a -> Text
+intercalateMap between showT = T.intercalate between . mapAndList showT
```

Comment:
Also, relaxing constraints for `mapAndList` function will allow to relax constraints for the `intercalateMap` function which will this function from using `[a]` to using some `Foldable f => f a` and this is an idiomatic generalisation for such kind of things.